floodlight/io/catapultApi.py

```diff
@@ -96,87 +111,123 @@ def _getAllplayerFormActivity_response(baseUrl: str, api_token: str, activity_id
     }
     endpoint = f"activities/{activity_id}/athletes/"
     response = requests.get(baseUrl + endpoint, headers=headers)
-    if response.status_code == 200:
-        return response
+    response_ok =200
+    if response.status_code == response_ok:
+        response_data = response.json()
+        
+        return response_data, endpoint
     else:
         raise APIRequestError(f"API request failed with status code {response.status_code}")
 
-def getResponse_data_json_list(baseUrl: str, api_token: str, activity_id: str) -> List[Dict]:
+def getResponse_data_dict_list(baseUrl: str, api_token: str, activity_id: str, response: List[Dict]) -> List[Dict]:
     """
     Retrieves sensor data for each player involved in a specific activity.
 
-    Args:
-        baseUrl (str): The base URL for the API requests.
-        api_token (str): The API token for authentication.
-        activity_id (str): The unique identifier for the activity.
-
-    Returns:
-        List[Dict]: A list of JSON data responses for each player, containing sensor data.
-
-    Raises:
-        APIRequestError: If any API request fails with a status code other than 200.
+    Parameters
+    ----------
+    baseUrl : str
+        The base URL for the API requests.
+    api_token : str
+        The API token for authentication.
+    activity_id : str
+        The unique identifier for the activity.
+    response : List[Dict]
+        The response containing player data.
+
+    Returns
+    -------
+    List[Dict]
+        A list of response_data for each player, containing sensor data.
+
+    Raises
+    ------
+    APIRequestError
+        If any API request fails with a status code other than 200.
     """
     headers = {
         "accept": "application/json",
         "authorization": f"Bearer {api_token}",
         "nulls": "0"
     }
-    allPlayersFromActivity_response = _getAllplayerFormActivity_response(baseUrl, api_token, activity_id)
-    allPlayersFromActivity_JSON = allPlayersFromActivity_response.json()
-    playerIDsInActivity_list = list(set(i["id"] for i in allPlayersFromActivity_JSON))
-    response_data_json_list = []
 
-    for i in playerIDsInActivity_list:
+    player_ids = list(set(i["id"] for i in response))

```

Comment:
I think transforming the `player_ids` into a list is not necessary as `player_ids.sort()` will do that anyways

Reply:
done